std/experimental/allocator/common.d

```diff
@@ -743,3 +743,135 @@ version(unittest)
         }
     }
 }
+
+/+
+Can the representation be determined at compile time to consist of nothing but
+zero bits? Padding between a struct's fields is not considered.
++/
+private template isAllZeroBits(T, T value)
+{
+    import std.traits : isAggregateType, isDynamicArray, isStaticArray;
+    static if (isDynamicArray!(typeof(value)))
+        enum isAllZeroBits = value is null && value.length == 0;
+    else static if (is(typeof(value is null)))
+        enum isAllZeroBits = value is null;
+    else static if (is(typeof(value is 0)))
+        enum isAllZeroBits = value is 0;
+    else static if (isStaticArray!(typeof(value)))
+    {
+        static if (value.length == 0)
+            enum isAllZeroBits = true;
+        else
+        {
+            alias E = typeof(value[0]);
+            enum isAllZeroBits = mixin(()
+            {
+                enum N = typeof(value).length;
+                char[] buffer = new char[32 * value.length]; // Pre-size for value.length <= 999.
```

Comment:
This doesn't reserve the size. On append the slice is not the tail so it wil reallocate.

Reply:
Thanks for the correction! `reserve` isn't available at compile time so I'll just remove this.